crates/bevy_ecs/src/lib.rs

```diff
@@ -1699,6 +1699,114 @@ mod tests {
         );
     }
 
+    #[test]
+    fn insert_batch() {
```

Comment:
could we get a test where you're inserting onto multiple entities in the same archetype?

Reply:
Is this good? (it passes) Also, should I make more for the other variations, or is one enough?
```
#[test]
fn insert_batch_same_archetype() {
    let mut world = World::default();
    let e0 = world.spawn((A(0), B(0))).id();
    let e1 = world.spawn((A(0), B(0))).id();
    let e2 = world.spawn(B(0)).id();

    let values = vec![(e0, (B(1), C)), (e1, (B(2), C)), (e2, (B(3), C))];

    world.insert_batch(values);

    assert_eq!(
        world.get::<A>(e0),
        Some(&A(0)),
        "first entity's A component should not have changed"
    );
    assert_eq!(
        world.get::<B>(e0),
        Some(&B(1)),
        "first entity's B component should have been replaced"
    );
    assert_eq!(
        world.get::<C>(e0),
        Some(&C),
        "first entity should have received C component"
    );
    assert_eq!(
        world.get::<A>(e1),
        Some(&A(0)),
        "second entity's A component should not have changed"
    );
    assert_eq!(
        world.get::<B>(e1),
        Some(&B(2)),
        "second entity's B component should have been replaced"
    );
    assert_eq!(
        world.get::<C>(e1),
        Some(&C),
        "second entity should have received C component"
    );
    assert_eq!(
        world.get::<A>(e2),
        None,
        "third entity should not have received A component"
    );
    assert_eq!(
        world.get::<B>(e2),
        Some(&B(3)),
        "third entity's B component should have been replaced"
    );
    assert_eq!(
        world.get::<C>(e2),
        Some(&C),
        "third entity should have received C component"
    );
}
```

Reply:
I think just this one test is good enough. The asserts are getting a little long. Would it be cleaner to use a query and collect the results?

Reply:
Much shorter:
```
#[test]
fn insert_batch_same_archetype() {
    let mut world = World::default();
    let e0 = world.spawn((A(0), B(0))).id();
    let e1 = world.spawn((A(0), B(0))).id();
    let e2 = world.spawn(B(0)).id();

    let values = vec![(e0, (B(1), C)), (e1, (B(2), C)), (e2, (B(3), C))];

    world.insert_batch(values);
    let mut query = world.query::<(Option<&A>, &B, &C)>();
    let component_values = query.get_many(&world, [e0, e1, e2]).unwrap();

    assert_eq!(
        component_values, 
        [(Some(&A(0)), &B(1), &C), (Some(&A(0)), &B(2), &C), (None, &B(3), &C)],
        "all entities should have had their B component replaced, received C component, and had their A component (or lack thereof) unchanged"
    );
}
```